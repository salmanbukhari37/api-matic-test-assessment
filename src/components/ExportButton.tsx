import React from "react";
import Button from "@mui/material/Button";

interface ExportButtonProps {
  onClick: () => void;
  disabled?: boolean;
}

const ExportButton: React.FC<ExportButtonProps> = ({
  onClick,
  disabled = false,
}) => {
  return (
    <Button
      variant="contained"
      color="primary"
      fullWidth
      sx={{
        borderRadius: "15px",
        fontWeight: "bold",
        backgroundColor: "#007bff",
        "&:hover": {
          backgroundColor: "#0056b3",
        },
      }}
      onClick={onClick}
      disabled={disabled}
    >
      Export
    </Button>
  );
};

export default ExportButton;
