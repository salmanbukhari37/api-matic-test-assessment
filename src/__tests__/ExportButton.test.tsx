import { render, screen, fireEvent } from "@testing-library/react";
import ExportButton from "../components/ExportButton";

describe("ExportButton Component", () => {
  const mockOnClick = jest.fn();

  test("renders correctly and matches snapshot", () => {
    const { container } = render(<ExportButton onClick={mockOnClick} />);
    expect(screen.getByRole("button", { name: /export/i })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("calls onClick handler when clicked", () => {
    render(<ExportButton onClick={mockOnClick} />);

    const button = screen.getByRole("button", { name: /export/i });
    fireEvent.click(button);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  test("is disabled when disabled prop is true", () => {
    render(<ExportButton onClick={mockOnClick} disabled={true} />);

    const button = screen.getByRole("button", { name: /export/i });
    expect(button).toBeDisabled();

    fireEvent.click(button);
    expect(mockOnClick).not.toHaveBeenCalled();
  });
});
