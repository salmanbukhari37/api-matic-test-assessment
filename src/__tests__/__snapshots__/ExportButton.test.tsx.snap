// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportButton Component renders correctly and matches snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-nnjf41-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Export
  </button>
</div>
`;
